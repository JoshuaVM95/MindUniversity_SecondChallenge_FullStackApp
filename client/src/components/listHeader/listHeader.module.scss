.listHeaderContainer {
	background: #333;
	color: #fff;
	display: grid;
	grid-template-columns: 1fr 4fr 1fr;
	height: 5rem;
	.listHeaderTitle {
		display: flex;
		align-items: center;
		justify-content: center;
		font-size: 2.2rem;
		font-weight: 500;
		text-align: center;
	}
	.filterIconContainer {
		display: flex;
		align-items: center;
		justify-content: flex-start;
		padding-left: 2rem;
		.filterIcon {
			font-size: 1.8rem;
		}
	}
	.addIconContainer {
		display: flex;
		align-items: center;
		justify-content: flex-end;
		padding-right: 2rem;
		.addIconButton {
			color: #fff;
			font-size: 1.8rem;
		}
	}
}
