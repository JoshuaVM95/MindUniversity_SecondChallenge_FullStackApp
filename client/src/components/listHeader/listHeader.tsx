import React from "react";
import { Box, Tooltip, IconButton } from "@material-ui/core";
import { FilterList } from "@material-ui/icons";
import styles from "./listHeader.module.scss";

interface ListHeaderProps {
	title: string;
	addIcon: React.ReactNode;
	addTooltipTitle: string;
	onAdd(): void;
	onOpenFilters?(): void;
	hasActiveFilters?: boolean;
}

export const ListHeader = ({
	title,
	addIcon,
	addTooltipTitle,
	onAdd,
	onOpenFilters,
	hasActiveFilters
}: ListHeaderProps): React.ReactElement => {
	return (
		<Box className={styles.listHeaderContainer}>
			<Box className={styles.filterIconContainer}>
				{onOpenFilters && (
					<Tooltip title="Open filter modal">
						<IconButton onClick={onOpenFilters} color={hasActiveFilters ? "info" : "inherit"}>
							<FilterList className={styles.filterIcon} />
						</IconButton>
					</Tooltip>
				)}
			</Box>

			<h1 className={styles.listHeaderTitle}>{title}</h1>
			<Box className={styles.addIconContainer}>
				<Tooltip title={addTooltipTitle}>
					<IconButton className={styles.addIconButton} onClick={onAdd}>
						{addIcon}
					</IconButton>
				</Tooltip>
			</Box>
		</Box>
	);
};
