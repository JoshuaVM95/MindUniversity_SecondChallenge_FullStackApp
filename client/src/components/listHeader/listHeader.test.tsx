import React from "react";
import { ListHeader } from "./listHeader";
import { render, fireEvent } from "@testing-library/react";
import { Add } from "@mui/icons-material";

describe("ListHeader tests", () => {
	const add = jest.fn();
	const renderListHeader = (openFilter?: () => void) => {
		return render(
			<ListHeader
				title="List header test"
				addIcon={<Add />}
				addTooltipTitle="Add test"
				onAdd={add}
				onOpenFilters={openFilter}
			/>
		);
	};
	it("renders correctly", () => {
		expect(renderListHeader().asFragment()).toMatchSnapshot();
	});
	it("renders the title prop correctly", () => {
		const listHeader = renderListHeader().asFragment();
		expect(listHeader).toHaveTextContent("List header test");
	});
	it("renders the Add icon", () => {
		const { getByTestId } = renderListHeader();
		const addIcon = getByTestId("AddIcon");
		expect(addIcon).toBeInTheDocument();
	});
	it("hiddes the filter button if there is no function", () => {
		const listHeader = renderListHeader().asFragment();
		expect(listHeader.querySelectorAll("button")).toHaveLength(1);
	});
	it("shows the filter button if the onOpenFilters function prop is send", () => {
		const listHeader = renderListHeader(() => console.log("onOpenFilters")).asFragment();
		expect(listHeader.querySelectorAll("button")).toHaveLength(2);
	});
	it("triggers the function when the filter button is click", () => {
		const openFilter = jest.fn();
		const { getByTestId } = renderListHeader(openFilter);
		const filterBtn = getByTestId("FilterListIcon");
		expect(filterBtn).toBeInTheDocument();
		fireEvent.click(filterBtn);
		expect(openFilter).toHaveBeenCalled();
	});
});
