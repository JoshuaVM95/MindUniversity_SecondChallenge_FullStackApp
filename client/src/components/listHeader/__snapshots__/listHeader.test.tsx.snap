// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader tests renders correctly 1`] = `
<DocumentFragment>
  <div
    class="listHeaderContainer MuiBox-root css-0"
  >
    <div
      class="filterIconContainer MuiBox-root css-0"
    />
    <h1
      class="listHeaderTitle"
    >
      List header test
    </h1>
    <div
      class="addIconContainer MuiBox-root css-0"
    >
      <button
        aria-label="Add test"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium addIconButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
